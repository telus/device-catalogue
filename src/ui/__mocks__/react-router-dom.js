import React from 'react';

const reactRouterDom = require('react-router-dom');
reactRouterDom.BrowserRouter = ({ children }) => <div>{children}</div>; // eslint-disable-line

module.exports = reactRouterDom;
