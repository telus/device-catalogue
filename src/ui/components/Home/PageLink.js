import React from 'react';
import Link from '../Common/LinkComponent';
import TdsLink from '@tds/core-link';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Image from '@tds/core-image';

const PageLink = ({ image, path, linkText, altText }) => {
  return (
    <Card variant="defaultOnlyBorder">
      <Box between={3} className="item-links">
        <Image src={image} alt={altText} width={90}/>
        <div>
          <Link TDSLink={TdsLink} to={path}>{linkText}</Link>
        </div>
      </Box>
    </Card>
  )
}

export default PageLink;