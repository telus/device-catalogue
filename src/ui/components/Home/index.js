import React from 'react';
import Heading from '@tds/core-heading';
import PageContainer from '../../wrappers/PageContainer';
import FlexGrid from '@tds/core-flex-grid';
import Box from '@tds/core-box';
import Image from '@tds/core-image';
import Card from '@tds/core-card';
import LinkWrapper from '../../wrappers/LinkWrapper';
import A11yContent from '@tds/core-a11y-content';
import iphoneImage from '../../../../public/images/iphone.jpg';
import watchImage from '../../../../public/images/watch.jpg';

const Home = () => (
  <main>
    <PageContainer>
      <section>
        <Heading level="h1">Device Catalogue</Heading>
        <Box vertical={5}>
          <FlexGrid>
            <FlexGrid.Row horizontalAlign="center">
              <FlexGrid.Col md={4}>
                <Box below={4}>
                  <Card variant="defaultWithBorder">
                    <FlexGrid.Row horizontalAlign="center">
                      <Heading level="h2">iPhones</Heading>
                    </FlexGrid.Row>
                    <FlexGrid.Row horizontalAlign="center">
                      <Box vertical={4}>
                        <Image
                          src={iphoneImage}
                          width={158}
                          height={200}
                          alt="iPhone"
                        />
                      </Box>
                    </FlexGrid.Row>
                    <FlexGrid.Row horizontalAlign="center">
                      <LinkWrapper button to="/iphones">
                        <span>
                          Shop <A11yContent>iPhones</A11yContent>Now
                        </span>
                      </LinkWrapper>
                    </FlexGrid.Row>
                  </Card>
                </Box>
              </FlexGrid.Col>
              <FlexGrid.Col md={4}>
                <Box below={4}>
                  <Card variant="defaultWithBorder">
                    <FlexGrid.Row horizontalAlign="center">
                      <Heading level="h2">Watches</Heading>
                    </FlexGrid.Row>
                    <FlexGrid.Row horizontalAlign="center">
                      <Box vertical={4}>
                        <Image
                          src={watchImage}
                          width={160}
                          height={200}
                          alt="watch"
                        />
                      </Box>
                    </FlexGrid.Row>
                    <FlexGrid.Row horizontalAlign="center">
                      <LinkWrapper button to="/watches">
                        <span>
                          Shop <A11yContent>watches</A11yContent>Now
                        </span>
                      </LinkWrapper>
                    </FlexGrid.Row>
                  </Card>
                </Box>
              </FlexGrid.Col>
            </FlexGrid.Row>
          </FlexGrid>
        </Box>
      </section>
    </PageContainer>
  </main>
);

export default Home;
