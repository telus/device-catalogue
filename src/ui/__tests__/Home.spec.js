import React from 'react';
import { render } from 'enzyme';
import Home from '../components/Home';
import { StaticRouter } from 'react-router-dom';

describe('Home page', () => {
  it('should render page with 2 links', () => {
    const wrapper = render(
      <StaticRouter>
        <Home />
      </StaticRouter>
    );

    const hrefs = Object.values(
      wrapper.find('a').map((idx, el) => el.attribs.href)
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('img').length).toEqual(2);
    expect(wrapper.text()).toMatch(/Device Catalogue/i);
    expect(hrefs).toContain('/iphones');
    expect(hrefs).toContain('/watches');
  });
});
