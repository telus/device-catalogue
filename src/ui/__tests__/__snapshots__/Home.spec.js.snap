// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render page with 2 links 1`] = `"<div class=\\"sc-bdvvtL eDFccf\\"><section><h1 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 fflItf\\">Device Catalogue</h1><div class=\\"sc-gsDKAQ sDFez\\"><div class=\\"FlexGrid__StyledGrid-sc-1da2tb5-0 tOBOA\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><div class=\\"Col__StyledCol-sc-15yvjc7-0 grnvOE\\"><div class=\\"sc-gsDKAQ dmSNiX\\"><div class=\\"sc-gsDKAQ jDAoVC sc-eCImPb chCNZB sc-eCImPb chCNZB\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><h2 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 hhjxPT\\">iPhones</h2></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><div class=\\"sc-gsDKAQ ipeOsr\\"><img src=\\"\\" width=\\"158\\" height=\\"200\\" alt=\\"iPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><a rank=\\"common\\" href=\\"/iphones\\" variant=\\"primary\\" class=\\"Button__StyledButton-sa7xo3-0 sc-gKclnd cTcNeS dWvYeV\\"><span>Shop <span class=\\"A11yContent__StyledA11yContent-sc-15wexwk-0 ftzZHp\\">iPhones</span>Now</span></a></div></div></div></div><div class=\\"Col__StyledCol-sc-15yvjc7-0 grnvOE\\"><div class=\\"sc-gsDKAQ dmSNiX\\"><div class=\\"sc-gsDKAQ jDAoVC sc-eCImPb chCNZB sc-eCImPb chCNZB\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><h2 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 hhjxPT\\">Watches</h2></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><div class=\\"sc-gsDKAQ ipeOsr\\"><img src=\\"\\" width=\\"160\\" height=\\"200\\" alt=\\"watch\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><a rank=\\"common\\" href=\\"/watches\\" variant=\\"primary\\" class=\\"Button__StyledButton-sa7xo3-0 sc-gKclnd cTcNeS dWvYeV\\"><span>Shop <span class=\\"A11yContent__StyledA11yContent-sc-15wexwk-0 ftzZHp\\">watches</span>Now</span></a></div></div></div></div></div></div></div></section></div>"`;
